### PR TITLE
Add custom email

### PR DESCRIPTION
**Pull request by anthonybaldwin**
- Add the ability to use non-random usernames
- Mac dying on the CURLs without 'k' flag; also made the CURLs match

### CODE REVIEW

**Comment by sdushantha** [comment]
I unfortunately have to disagree with this PR.

> Add the ability to use non-random usernames

If you were to use a username that is already taken, you would get emails of the person who as the same username. This would be a huge privacy risk.
But of course if you would like this feature, just make your own fork of `tmpmail` with this feature added.

> Mac dying on the CURLs without 'k' flag; also made the CURLs match

This issue was occuring because of an invalid certificate which 1secmail had. This issue has been fixed in #7 


Hope that makes sense :)

**Comment by anthonybaldwin** [comment]
Adding the ability to enter any username keeps the tool consistent with the GUI where you can do this. Inboxes are nuked periodically anyway (every 10 minutes?).

>Disposable email is a service, that allows users to generate email address and receive emails at this temporary address. After certain period of time email will be delated and address will be canceled.

If randomness and privacy is a concern when using a public tool like this, another option would be to enter a custom string, take X number of characters from it, then append random characters after that. With some encoding could do something like `zjXhGljhsDShd+customString@1secmail`. Very useful for testing IMO.

image

----

I actually saw the SSL error via the CURL and the browser; should've removed that bit... Doh.

**Comment by sdushantha** [comment]
I understand what you are saying. 

The reason why I use `tmpmail` is to receive verification codes from a website or when I need to give a website my email in order to view their content. So in my case, I just need an email address and the username does not matter for me.

Could you give me an example of why you would need a custom username? Because I cant think of a scenario where someone would need a specific username for their temporary email.

**Comment by anthonybaldwin** [comment]
I guess it just depends on the usage.

An example would be for QA working at a company where temporary email domains are disallowed (for their customers); hypothetically speaking, one could get around said block by having a "keyword bypass" added to whatever filtering system is in use for spam/freemail domains.

It's no big deal if you don't want to add the functionality. I understand where you're coming from as well, and it's easy for me to use a fork. :)

**Comment by sdushantha** [comment]
@anthonybaldwin I'm reopening this PR because @zanona was able to provide a good reason in #22 for why we should implement the custom username feature.

File: tmpmail [hunk 1]
```diff
@@ -34,7 +34,7 @@ usage(){
     # Using 'cat << EOF' we can easily output a multiline text. This is much
     # better than using 'echo' for each line or using '\n' to create a new line.
     cat << EOF
-usage: tmpmail [-h] [--generate] [--browser BROWSER] [--recent] ID
+usage: tmpmail [-h] [--generate] CUSTOM [--browser BROWSER] [--recent] ID
```
Comment (sdushantha) [inline tmpmail:37]: Could you replace `CUSTOM` with `USERNAME`?

**Comment by sdushantha** [comment]
If you fix the conflicts and remove the `-k` flag from `curl` I'll merge this PR

**Comment by sdushantha** [comment]
@anthonybaldwin I had some extra free time today, so I've taken care of this :)

**Pull request merged**